src/main/java/store/controller/StoreController.java

```diff
@@ -0,0 +1,117 @@
+        PromotionRepository promotionRepository = new MemoryPromotionRepository();
+        productService = new ProductService(productRepository, promotionRepository);
+        productStockHandler = new ProductStockHandler(productRepository, productService, receipt);
+        promotionService = new PromotionService(promotionRepository);
+
+        copyBackupFile();
+    }
+
+    private void copyBackupFile() {
+        try {
+            Files.copy(Path.of(PRODUCT_FILE_PATH), Path.of(BACKUP_FILE_PATH), StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            System.err.println(e.getMessage());
+        }
+    }
+
+    public void storeServiceRun() {
+        OutputView.printWelcomeMessage();
+        promotionService.loadPromotionsFromFile();
+        showProductsList();
+
+        while (true) {
+            try {
+                System.out.println();
+                Map<String, Integer> productsMap = chooseProductAndQuantity();
+                if (updateStockOnFile(productsMap)) {
+                    if (InputView.askMembershipDiscount()) {
+                        receipt.setMembershipDiscount();
+                    }
+                    receiptService = new ReceiptService(receipt);
+                    calculateTotalMoney();
+
+                    if (!InputView.askToContinueShopping()) {
+                        break;
+                    }
+                    receipt = new Receipt();
+                    productStockHandler.updateReceipt(receipt);
+
+                    OutputView.printWelcomeMessage();
+                    promotionService.loadPromotionsFromFile();
+                    showProductsList();
+                }
+            } catch (InputException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+    private void showProductsList() {
+        productService.loadProductsFromFile();
+    }
+
+    private Map<String, Integer> chooseProductAndQuantity() {
+        while (true) {
+            try {
+                OutputView.printProductNameAndQuantity();
+                String productNameAndQuantity = InputView.inputProductNameAndQuantity();
+                return ProductParser.parseProducts(productNameAndQuantity);
+            } catch (InputException e) {
+                System.out.println(e.getMessage());
+                System.out.println();
+            }
+        }
+    }
+
+    private boolean updateStockOnFile(Map<String, Integer> productsMap) throws InputException {
+        boolean isStockDeducted = false;
+        for (Map.Entry<String, Integer> entry : productsMap.entrySet()) {
+            try {
+                isStockDeducted = false;
+                productStockHandler.deductStock(entry.getKey(), entry.getValue());
+                isStockDeducted = true;
+            } catch (InputException e) {
+                System.out.println(e.getMessage());
+            }
+        }
```

Comment:
stream을 이용하면 더 깔끔해지는것 같습니다!
```suggestion
        productsMap.forEach((key, value) -> {
    try {
        isStockDeducted = false;
        productStockHandler.deductStock(key, value);
        isStockDeducted = true;
    } catch (InputException e) {
        System.out.println(e.getMessage());
    }
});
```

Reply:
오 코드 설명까지..!
감사합니다!
stream으로 `getKey`, `getValue`사용을 없앨 수 있었군요!